pandas/tests/groupby/test_any_all.py

```diff
@@ -68,3 +69,91 @@ def test_bool_aggs_dup_column_labels(bool_agg_func):
 
     expected = df
     tm.assert_frame_equal(result, expected)
+
+
+@pytest.mark.parametrize("bool_agg_func", ["any", "all"])
+@pytest.mark.parametrize("skipna", [True, False])
+@pytest.mark.parametrize(
+    # expected_data indexed as [[skipna=False/any, skipna=False/all],
+    #                           [skipna=True/any, skipna=True/all]]
+    "data,expected_data",
+    [
+        ([False, False, False], [[False, False], [False, False]]),
+        ([True, True, True], [[True, True], [True, True]]),
+        ([pd.NA, pd.NA, pd.NA], [[pd.NA, pd.NA], [False, True]]),
+        ([False, pd.NA, False], [[pd.NA, False], [False, False]]),
+        ([True, pd.NA, True], [[True, pd.NA], [True, True]]),
+        ([True, pd.NA, False], [[True, False], [True, False]]),
+    ],
+)
+def test_masked_kleene_logic(bool_agg_func, data, expected_data, skipna):
+    # GH#37506
+    df = DataFrame(data, dtype="boolean")
+    expected = DataFrame(
+        [expected_data[skipna][bool_agg_func == "all"]], dtype="boolean", index=[1]
+    )
+
+    result = df.groupby([1, 1, 1]).agg(bool_agg_func, skipna=skipna)
+    tm.assert_frame_equal(result, expected)
+
+    # The expected result we compared to should match aggregating on the whole
+    # series
+    result = getattr(df[0], bool_agg_func)(skipna=skipna)
+    expected = expected_data[skipna][bool_agg_func == "all"]
+    assert (result is pd.NA and expected is pd.NA) or result == expected
+
+
+@pytest.mark.parametrize(
+    "dtype1,dtype2,exp_col1,exp_col2",
+    [
+        (
+            "float",
+            "Float64",
+            pd.array([True], dtype=bool),
```

Comment:
```suggestion
            np.array([True], dtype=bool),
```

(for the resulting DataFrame, both are equivalent, but since the goal is to create a column with a numpy dtype, I find it a bit more clear / explicit to create a numpy array)

Reply:
Sounds good